ccl/multiregionccl: TestColdStartLatency failed

ccl/multiregionccl.TestColdStartLatency [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18674986?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18674986?buildTab=artifacts#/) on master @ [c9b028b69bb5de829374989a139356c7415920ad](https://github.com/cockroachdb/cockroach/commits/c9b028b69bb5de829374989a139356c7415920ad):


```
=== RUN   TestColdStartLatency
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/201913953f2c683da40d46e0bffd06a3/logTestColdStartLatency3224965048
    test_log_scope.go:76: use -show-logs to present logs inline
    cold_start_latency_test.go:181: 0 alter range meta configure zone using constraints = '{"+region=us-east1": 1, "+region=us-west1": 1, "+region=europe-west1": 1}';
    cold_start_latency_test.go:181: 0 
        BEGIN;
        ALTER DATABASE system PRIMARY REGION "us-east1";
        ALTER DATABASE system ADD REGION "us-west1";
        ALTER DATABASE system ADD REGION "europe-west1";
        COMMIT;
    cold_start_latency_test.go:182: error executing query="\nBEGIN;\nALTER DATABASE system PRIMARY REGION \"us-east1\";\nALTER DATABASE system ADD REGION \"us-west1\";\nALTER DATABASE system ADD REGION \"europe-west1\";\nCOMMIT;" args=[]: pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE - failed preemptive refresh - unknown error: TransactionAbortedError(ABORT_REASON_ABORT_SPAN): "sql txn" meta={id=9b01cb52 key=/Tenant/10/Table/3/1/3/2/1 iso=Serializable pri=0.00977251 epo=0 ts=1738280531.175795582,2 min=1738280530.668224875,0 seq=864} lock=true stat=ABORTED rts=1738280531.175795582,2 wto=false gul=1738280531.168224875,0): "sql txn" meta={id=9b01cb52 key=/Tenant/10/Table/3/1/3/2/1 iso=Serializable pri=0.00977251 epo=0 ts=1738280531.175795582,2 min=1738280530.668224875,0 seq=864} lock=true stat=PENDING rts=1738280530.668224875,0 wto=false gul=1738280531.168224875,0
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/201913953f2c683da40d46e0bffd06a3/logTestColdStartLatency3224965048
--- FAIL: TestColdStartLatency (8.28s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestColdStartLatency.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-47022

ccl/multiregionccl.TestColdStartLatency [failed](https://tanzanite.cluster.engflow.com/invocations/default/3788a372-18fe-437a-bb95-ae8d901147e8?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL211bHRpcmVnaW9uY2NsOm11bHRpcmVnaW9uY2NsX3Rlc3Q=) on master @ [5b931856d005214057f0703890ce3a77cdc21f30](https://github.com/cockroachdb/cockroach/commits/5b931856d005214057f0703890ce3a77cdc21f30):


```
=== RUN   TestColdStartLatency
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestColdStartLatency1741753963
    test_log_scope.go:76: use -show-logs to present logs inline
    cold_start_latency_test.go:181: 0 alter range meta configure zone using constraints = '{"+region=us-east1": 1, "+region=us-west1": 1, "+region=europe-west1": 1}';
    cold_start_latency_test.go:181: 0 
        BEGIN;
        ALTER DATABASE system PRIMARY REGION "us-east1";
        ALTER DATABASE system ADD REGION "us-west1";
        ALTER DATABASE system ADD REGION "europe-west1";
        COMMIT;
    cold_start_latency_test.go:182: error executing query="\nBEGIN;\nALTER DATABASE system PRIMARY REGION \"us-east1\";\nALTER DATABASE system ADD REGION \"us-west1\";\nALTER DATABASE system ADD REGION \"europe-west1\";\nCOMMIT;" args=[]: pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_ASYNC_WRITE_FAILURE - missing intent on: /Tenant/10/Table/54/1/1042878744714182657/"legacy_payload"/1914-12-01T10:04:23.506620999Z/0): "sql txn" meta={id=daa728e8 key=/Tenant/10/Table/3/1/56/2/1 iso=Serializable pri=0.03343555 epo=0 ts=1738331736.406080195,1 min=1738331735.493378559,0 seq=0} lock=true stat=PENDING rts=1738331736.406080195,1 wto=false gul=1738331735.993378559,0
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestColdStartLatency1741753963
--- FAIL: TestColdStartLatency (14.59s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestColdStartLatency.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/multiregionccl/cold_start_latency_test.go
+++ b/pkg/ccl/multiregionccl/cold_start_latency_test.go
@@ -171,6 +171,7 @@ func TestColdStartLatency(t *testing.T) {
 		} else {
 			stmts = []string{`
 BEGIN;
+SET LOCAL autocommit_before_ddl = false;
 ALTER DATABASE system PRIMARY REGION "us-east1";
 ALTER DATABASE system ADD REGION "us-west1";
 ALTER DATABASE system ADD REGION "europe-west1";

